cmd/commands/datasetIngestor.go

```diff
@@ -237,25 +237,30 @@ For Windows you need instead to specify -user username:password on the command l
 		// a destination location is defined by the archive system
 		// for now let the user decide if he needs a copy
 
-		// now everything is prepared, prepare to loop over all folders
 		if nocopyFlag {
 			copyFlag = false
 		}
 		checkCentralAvailability := !(cmd.Flags().Changed("copy") || cmd.Flags().Changed("nocopy") || beamlineAccount || copyFlag)
-		skip := ""
+		skipSymlinks := ""
 
 		// check if skip flag is globally defined via flags:
 		if cmd.Flags().Changed("linkfiles") {
 			switch linkfiles {
 			case "delete":
-				skip = "sA"
+				skipSymlinks = "sA"
 			case "keep":
-				skip = "kA"
+				skipSymlinks = "kA"
 			default:
-				skip = "dA" // default behaviour = keep internal for all
+				skipSymlinks = "dA" // default behaviour = keep internal for all
 			}
 		}
 
+		var skippedLinks uint = 0
```

Comment:
the fact that file wide variables are changed in place feels a little prone to errors (since it's not idempotent). Given that you are refactoring GetLocalFileList, why not make it return these 2 extra values?

Reply:
due to the way Go handles stuff, it was package wide. I can return a number for the current dataset, but not for the whole list. This can be added to a variable local to the ingestor function to keep track of the total (the original behaviour). Even if I did not use callbacks, I'd still need this variable.

Reply:
And global variables ("file wide") are almost never a good idea, they'll certainly cause issues in a threaded environment (eg. REST server). Normally the variable should be separate for each ingestion process, which can only be done (or the easiest way at least) if the variable is local to the function call.

Reply:
sorry, my comment was badly written. See [here](https://github.com/paulscherrerinstitute/scicat-cli/pull/104#discussion_r1731038141) what I meant